dotnet-statsig/src/Statsig/Network/EventLogger.cs

```diff
@@ -48,17 +48,11 @@ public void Enqueue(EventLog entry)
             _eventLogQueue.Add(entry);
             if (_eventLogQueue.Count >= _maxQueueLength)
             {
-                ForceFlush();
+                await FlushEvents();
```

Comment:
This was meant to be a fire and forget action, so that the actual `enqueue` of events is not blocking. Do we need to make this `await` for any reason?

Reply:
Without the await call, then any exceptions thrown within FlushEvents will not be handled, which can result in a similar application crash as the one I contacted you about earlier today.  If you want it to be a fire-and-forget operation, then you should do something like this in order to make _sure_ that there are no unhandled exceptions.
```cs
Task.Run(async () =>
{
    try
    {
        await FlushEvents();
    }
    catch (Exception ex)
    {
        // log, etc.
    }
});
```

Reply:
I also didn't realize that I accidentally made the Enqueue method async.  `async void` is pretty sketchy as it also means that nothing will ever be able to catch and handle thrown exceptions.  I must have accidentally changed that method to be async as I was reviewing the code and looking at what would need to change, and then later when I added the await on FlushEvents() here, and it didn't complain, I just assumed the method was already returning a Task.  I'll fix this by dropping the `async` keyword from this method, and then add the Task.Run(...) wrapper I mentioned above